go/vt/vtgate/engine/sequential.go

```diff
@@ -0,0 +1,90 @@
+/*
+Copyright 2023 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package engine
+
+import (
+	"context"
+
+	"vitess.io/vitess/go/sqltypes"
+	querypb "vitess.io/vitess/go/vt/proto/query"
+	vtrpcpb "vitess.io/vitess/go/vt/proto/vtrpc"
+	"vitess.io/vitess/go/vt/vterrors"
+)
+
+// Sequential Primitive is used to execute DML statements in a fixed order.
+// Any failure, stops the execution and returns.
+type Sequential struct {
+	Sources []Primitive
+
+	txNeeded
+}
+
+var _ Primitive = (*Sequential)(nil)
+
+// NewSequential creates a Sequential primitive.
+func NewSequential(Sources []Primitive) *Sequential {
+	return &Sequential{
+		Sources: Sources,
+	}
+}
+
+// RouteType returns a description of the query routing type used by the primitive
+func (s *Sequential) RouteType() string {
+	return "Sequential"
+}
+
+// GetKeyspaceName specifies the Keyspace that this primitive routes to
+func (s *Sequential) GetKeyspaceName() string {
+	res := s.Sources[0].GetKeyspaceName()
+	for i := 1; i < len(s.Sources); i++ {
+		res = formatTwoOptionsNicely(res, s.Sources[i].GetKeyspaceName())
+	}
+	return res
+}
+
+// GetTableName specifies the table that this primitive routes to.
+func (s *Sequential) GetTableName() string {
+	res := s.Sources[0].GetTableName()
+	for i := 1; i < len(s.Sources); i++ {
+		res = formatTwoOptionsNicely(res, s.Sources[i].GetTableName())
+	}
+	return res
+}
+
+// TryExecute performs a non-streaming exec.
+func (s *Sequential) TryExecute(ctx context.Context, vcursor VCursor, bindVars map[string]*querypb.BindVariable, wantFields bool) (*sqltypes.Result, error) {
+	return nil, vterrors.VT10002()
```

Comment:
what am I missing here? why error on both execute methods?

Reply:
`replace into` plan execution is disabled as of now as there is an inherent issue with locking rows with `unique` keys.

Reply:
This exposed an issue with mysql locking with unique key. A follow up PR will add the support for it on the engine.